Aquib/1_Valid_Sudoku.c.txt

```diff
@@ -0,0 +1,83 @@
+bool checkRow(char **board, int n)
+{
+    char boardstatus[10] = {0};
```

Comment:
Why is the size 10? Shouldn't it be 9?

Reply:
In C, a character array "char boardstatus[10]" used above is a string. Every string ends with a "\0" that is not used but only used to denote the end of the string. If I put 9 as the size of the char array, I will only have 8 elements that I can access in a loop. Since I need to iterate over 9 rows and columns and accordingly fill in the above array, I have to allocate an extra space for the "\0" , hence I allocated memory for 10 elements in that array.

Reply:
Right, makes sense!